llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp

```diff
@@ -886,34 +902,40 @@ bool SIInsertWaitcnts::applyPreexistingWaitcnt(
     if (II.isMetaInstruction())
       continue;
 
-    if (II.getOpcode() == AMDGPU::S_WAITCNT) {
+    unsigned Opcode = II.getOpcode();
+    bool CanFullyDiscardWaitcntSequence = SIInstrInfo::isSoftWaitcnt(Opcode);
+
+    if (SIInstrInfo::isWaitcnt(Opcode)) {
       // Conservatively update required wait if this waitcnt was added in an
       // earlier pass. In this case it will not exist in the tracked waitcnt
       // set.
```

Comment:
What should it say instead?

Reply:
Well, there's no more tracked waitcnt set, so we should rephrase it in terms of soft waitcnts instead.

Something along the lines of "Update required wait. If this waitcnt was added in an earlier pass, but is no longer needed, it may be removed."